src/vm/appdomain.cpp

```diff
@@ -12583,6 +12583,8 @@ AppDomain::RaiseAssemblyResolveEvent(
             if ( pSpec->IsIntrospectionOnly() 
 #ifdef FEATURE_FUSION
                     || pSpec->GetParentLoadContext() == LOADCTX_TYPE_UNKNOWN
+#elif FEATURE_CORECLR
+                    || pSpec->GetBindingContextFromParentAssembly(this) != GetTPABinderContext()
 #endif
```

Comment:
Ok, I have not realized that this is null most of the time anyway ... it looked to me like that this code tries to make it artificially null. Maybe worth a comment.